src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -110,14 +110,34 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
                 logging.warning(msg)
                 results.put({'name': name, 'success': "error", 'error': msg})
             else:
-                msg = "Error trying to process block %s through DBS. Error: %s" % (name, exString)
+                reason = parseDBSException(exString)
```

Comment:
Valentin, I wonder if we should actually move this line up to L95, basically replacing this:
```
exString = str(getattr(ex, "body", ex))
```
?

Would we be able to find a message like "Block BLAH already exists" in the error `reason` attribute? Or is it only in the `body`?

Reply:
The DBS server `reason` does not provide this information, it will provide reason why certain API and its transaction fails, e.g. if block exists in DB the `reason` will provide ORACLE exact error. Moreover, for POST HTTP requests which may have O(100MB) payload I doubt that posting back content to the `reason` is useful. That's why I settled that error should only provide reason, while client who send request knows which request was sent to DBS server and the client knows which parameters were used in HTTP request.

Regarding entire block. It depends:
- this statement
```
if 'Block %s already exists' % name in exString:
```
comes from DB Python server
- this one
```
elif 'Proxy Error' in exString:
```
comes from FE when it can't reach DBS BE
- and this one
```
elif 'Missing data when inserting to dataset_parents' in exString:
```
comes from Python server too

Therefore, you have few options:
- leave python related ifs for backward compatibility or remove them
- replaces all ifs except last one I provided since they `parseDBSException` will provide body if it can't parse it.

Please clearly specify how to proceed and I can adjust the code accordingly.

Reply:
I wonder if this development can be impacted by the server-side changes discussed here:
https://github.com/dmwm/dbs2go/issues/48#issuecomment-1152937048
?

Maybe let's get that one fixed and tested first, then we come back to this one.

Reply:
The way I added my changes will work fine with https://github.com/dmwm/dbs2go/issues/48#issuecomment-1152937048 since I preserved initial if/else flow. As such first condition in that flow is checking for Block message if it exists. But I totally agree that this PR we should apply after we put fix for Block message into production.

Reply:
Valentin, I think all these if/else blocks need to remain because we have special actions (and logging!) based on the kind of error reported by the server.
However, I would suggest to call this `parseDBSException` on the exception object at the very first line of the except block, thus replacing this:
```
exString = str(getattr(ex, "body", ex))
```

All the if/else checks will remain unchanged and functional (comparing against a smaller version of what we used to have in exString

Reply:
Marking this as resolved